src/main/kotlin/com/team4099/robot2023/subsystems/superstructure/Superstructure.kt

```diff
@@ -0,0 +1,990 @@
+          )
+        if (groundIntake.isAtTargetedPosition) {
+          manipulator.currentRequest =
+            Request.ManipulatorRequest.TargetingPosition(
+              Manipulator.TunableManipulatorStates.groundIntakeConeExtension.get(),
+              ManipulatorConstants.IDLE_VOLTAGE
+            )
+        }
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition && manipulator.isAtTargetedPosition) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CONE
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CONE_CLEANUP
+        }
+      }
+      SuperstructureStates.GROUND_INTAKE_CONE -> {
+        // Outputs
+        manipulator.currentRequest =
+          Request.ManipulatorRequest.TargetingPosition(
+            Manipulator.TunableManipulatorStates.groundIntakeConeExtension.get(),
+            Manipulator.TunableManipulatorStates.coneInVoltage.get()
+          )
+
+        // Transition
+        if (manipulator.isAtTargetedPosition) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CONE_CLEANUP
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.GROUND_INTAKE_CONE_CLEANUP
+        }
+      }
+      SuperstructureStates.GROUND_INTAKE_CONE_CLEANUP -> {
+        // Outputs
+        manipulator.currentRequest =
+          Request.ManipulatorRequest.TargetingPosition(
+            Manipulator.TunableManipulatorStates.minExtension.get(),
+            Manipulator.TunableManipulatorStates.coneIdleVoltage.get()
+          )
+
+        // Transition
+        if (manipulator.isAtTargetedPosition) {
+          nextState = SuperstructureStates.IDLE
+        }
+      }
+      SuperstructureStates.DOUBLE_SUBSTATION_INTAKE_PREP -> {
+        // Outputs
+        groundIntake.currentRequest =
+          Request.GroundIntakeRequest.TargetingPosition(
+            GroundIntake.TunableGroundIntakeStates.stowedDownAngle.get(),
+            GroundIntake.TunableGroundIntakeStates.neutralVoltage.get()
+          )
+        if (groundIntake.isAtTargetedPosition) {
+          manipulator.currentRequest =
+            Request.ManipulatorRequest.TargetingPosition(
+              Manipulator.TunableManipulatorStates.doubleSubstationIntakeShelfExtension.get(),
+              ManipulatorConstants.IDLE_VOLTAGE
+            )
+          val offset =
+            when (usingGamePiece) {
+              GamePiece.CUBE -> Elevator.TunableElevatorHeights.shelfIntakeCubeOffset.get()
+              GamePiece.CONE -> Elevator.TunableElevatorHeights.shelfIntakeConeOffset.get()
+              else -> 0.0.inches
+            }
+          elevator.currentRequest =
+            Request.ElevatorRequest.TargetingPosition(
+              Elevator.TunableElevatorHeights.doubleSubstationHeight.get() + offset
+            )
+        }
+
+        // Transition
+        if (groundIntake.isAtTargetedPosition &&
```

Comment:
is it possible that this condition fires before we want it to? like is it possible that the ground intake is at its position but the manipulator and elevator have not received an updated setpoint, so all 3 are "in position"? 

consider what happens if in between the first check for `isAtTargetedPosition` and this line, the motor controller/gravity/whatever cause the intake to move into position.